app/assets/javascripts/app/views/publisher/mention_view.js

```diff
@@ -0,0 +1,328 @@
+
+    this.addMention(mention);
+
+    // Cleaning before inserting the value, otherwise auto-complete would be triggered with "old" inputbuffer
+    this.clearBuffer();
+    this.currentDataQuery = "";
+    this.resetMentionBox();
+
+    // Mentions & syntax message
+    var updatedMessageText = start + this.mentionChar + mention.name + end;
+    this.elmInputBox.val(updatedMessageText);
+    this.updateValues();
+
+    // Set correct focus and selection
+    this.elmInputBox.focus();
+    this.setCaretPosition(startEndIndex);
+  },
+
+  updateValues: function(){
+    var syntaxMessage = this.getInputBoxValue();
+    var mentionText = this.getInputBoxValue();
+    this.clearFilteredResults();
+
+    var self = this;
+
+    _.each(this.mentionsCollection, function(mention){
+      self.addToFilteredResults(mention);
+
+      var mentionVal = self.mentionChar + mention.name;
+
+      var textSyntax = self.settings.templates.mentionItemSyntax(mention);
+      syntaxMessage = syntaxMessage.replace(mentionVal, textSyntax);
+
+      var textHighlight = self.settings.templates.mentionItemHighlight({name: _.escape(mention.name)});
+      mentionText = mentionText.replace(mentionVal, textHighlight);
+    });
+
+    mentionText = mentionText.replace(/\n/g, "<br/>");
+    mentionText = mentionText.replace(/ {2}/g, "&nbsp; ");
+
+    this.elmInputBox.data("messageText", syntaxMessage);
+    this.elmMentionsOverlay.find("div > div").html(mentionText);
+  },
+
+  /**
+   * Let us prefill the publisher with a mention list
+   * @param persons List of people to mention in a post;
+   * JSON object of form { handle: <diaspora handle>, name: <name>, ... }
+   */
+  prefillMention: function(persons){
+    var self = this;
+    _.each(persons, function(person){
+      self.addMention(person);
+      self.addToFilteredResults(person);
+      var text = self.mentionChar + person.name;
+      if(self.elmInputBox.val().length !== 0){
+        text = self.elmInputBox.val() + " " + text;
+      }
+      self.elmInputBox.val(text);
+      self.updateValues();
+    });
+  },
+
+  selectNextResult: function(evt){
+    if(this.isVisible()){
+      evt.preventDefault();
+      evt.stopPropagation();
+    }
+
+    if(this.getSelected().size() !== 1 || this.getSelected().next().size() !== 1){
+      this.getSelected().removeClass("tt-cursor");
+      this.$el.find(".tt-suggestion").first().addClass("tt-cursor");
+    }
+    else{
+      this.getSelected().removeClass("tt-cursor").next().addClass("tt-cursor");
+    }
+  },
+
+  selectPreviousResult: function(evt){
+    if(this.isVisible()){
+      evt.preventDefault();
+      evt.stopPropagation();
+    }
+
+    if(this.getSelected().size() !== 1 || this.getSelected().prev().size() !== 1){
+      this.getSelected().removeClass("tt-cursor");
+      this.$el.find(".tt-suggestion").last().addClass("tt-cursor");
+    }
+    else{
+      this.getSelected().removeClass("tt-cursor").prev().addClass("tt-cursor");
+    }
+  },
```

Comment:
I did this manually for the search input on develop:
```js
$("header .tt-input").typeahead("val", "te");
$("header .tt-input").typeahead("activate");
$("header .tt-input").typeahead("open");
$("header .tt-input").trigger($.Event("keydown", {keyCode:40}));
```
You should be able to do something similar in this PR.